modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/security/service/access/policy/CaptchaSAPEntryPortalInstanceLifecycleListener.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.captcha.rest.internal.security.service.access.policy;
+
+import com.liferay.portal.instance.lifecycle.BasePortalInstanceLifecycleListener;
+import com.liferay.portal.instance.lifecycle.PortalInstanceLifecycleListener;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.language.LanguageResources;
+import com.liferay.portal.security.service.access.policy.model.SAPEntry;
+import com.liferay.portal.security.service.access.policy.service.SAPEntryLocalService;
+
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Quan Huynh
+ */
+@Component(service = PortalInstanceLifecycleListener.class)
+public class CaptchaSAPEntryPortalInstanceLifecycleListener
+	extends BasePortalInstanceLifecycleListener {
+
+	@Override
+	public void portalInstancePreregistered(Company company) throws Exception {
+		try {
+			_addSAPEntry(company.getCompanyId());
+		}
+		catch (PortalException portalException) {
+			_log.error(
+				"Unable to add service access policy entry for company " +
+					company.getCompanyId(),
+				portalException);
+		}
+	}
+
+	private void _addSAPEntry(long companyId) throws PortalException {
+		String name = _SAP_ENTRY_OBJECT_ENTRY[0];
+
+		SAPEntry sapEntry = _sapEntryLocalService.fetchSAPEntry(
+			companyId, name);
+
+		if (sapEntry != null) {
+			return;
+		}
+
+		Map<Locale, String> map = ResourceBundleUtil.getLocalizationMap(
+			LanguageResources.PORTAL_RESOURCE_BUNDLE_LOADER,
+			"service-access-policy-entry-default-captcha-entry-title");
+
+		_sapEntryLocalService.addSAPEntry(
+			_userLocalService.getGuestUserId(companyId),
+			_SAP_ENTRY_OBJECT_ENTRY[1], true, true, name, map,
+			new ServiceContext());
+	}
+
+	private static final String[] _SAP_ENTRY_OBJECT_ENTRY = {
```

Comment:
should be `_SAP_ENTRY_OBJECT_ARRAY` instead of?

Reply:
@ces-quanhuynh how about this?

Reply:
Fixed it. Thanks